packages/apollo-server-koa/src/ApolloServer.ts

```diff
@@ -0,0 +1,178 @@
+
+export interface ServerRegistration {
+  // Note: You can also pass a connect.Server here. If we changed this field to
+  // `express.Application | connect.Server`, it would be very hard to get the
+  // app.use calls to typecheck even though they do work properly. Our
+  // assumption is that very few people use connect with TypeScript (and in fact
+  // we suspect the only connect users left writing GraphQL apps are Meteor
+  // users).
+  app: Koa;
+  path?: string;
+  cors?: corsMiddleware.Options | boolean;
+  bodyParserConfig?: Object | boolean;
+  onHealthCheck?: (ctx: Koa.Context) => Promise<any>;
+  disableHealthCheck?: boolean;
+  gui?: boolean;
+}
+
+const fileUploadMiddleware = (
+  uploadsConfig: FileUploadOptions,
+  server: ApolloServerBase,
+) => (ctx: Koa.Context, next: Function) => {
+  if (typeis(ctx.req, ['multipart/form-data'])) {
+    return processFileUploads(ctx.req, uploadsConfig)
+      .then(body => {
+        ctx.request.body = body;
+        return next();
+      })
+      .catch(error => {
+        if (error.status && error.expose) ctx.status = error.status;
+
+        throw formatApolloErrors([error], {
+          formatter: server.requestOptions.formatError,
+          debug: server.requestOptions.debug,
+        });
+      });
+  } else {
+    return next();
+  }
+};
+
+export class ApolloServer extends ApolloServerBase {
+  // This translates the arguments from the middleware into graphQL options It
+  // provides typings for the integration specific behavior, ideally this would
+  // be propagated with a generic to the super class
+  async createGraphQLServerOptions(ctx: Koa.Context): Promise<GraphQLOptions> {
+    return super.graphQLServerOptions(ctx);
+  }
+
+  protected supportsSubscriptions(): boolean {
+    return true;
+  }
+
+  protected supportsUploads(): boolean {
+    return true;
+  }
+
+  public applyMiddleware({
+    app,
+    path,
+    cors,
+    bodyParserConfig,
+    disableHealthCheck,
+    gui,
+    onHealthCheck,
+  }: ServerRegistration) {
+    const router = new KoaRouter();
+
+    if (!path) path = '/graphql';
+
+    if (!disableHealthCheck) {
+      // uses same path as engine proxy, but is generally useful.
+      router.all('/.well-known/apollo/server-health', (ctx: Koa.Context) => {
+        // Response follows https://tools.ietf.org/html/draft-inadarei-api-health-check-01
+        ctx.set('Content-Type', 'application/health+json');
+
+        if (onHealthCheck) {
+          return onHealthCheck(ctx)
+            .then(() => {
+              ctx.body = { status: 'pass' };
+            })
+            .catch(() => {
+              ctx.status = 503;
+              ctx.body = { status: 'fail' };
+            });
+        } else {
+          ctx.body = { status: 'pass' };
+        }
+      });
+    }
+
+    let uploadsMiddleware;
+    if (this.uploadsConfig) {
+      uploadsMiddleware = fileUploadMiddleware(this.uploadsConfig, this);
+    }
+
+    // XXX multiple paths?
```

Comment:
This was probably left over from the express integration. We should remove this, since `applyMiddleware` can be called multiple times with different paths.